components/view_image/view_image.jsx

```diff
@@ -262,6 +269,18 @@ export default class ViewImageModal extends React.PureComponent {
             );
         }
 
+        for (const preview of this.props.pluginFilePreviewComponents) {
+            if (preview.override(fileInfo, this.props.post)) {
```

Comment:
What happens if I have multiple plugins trying to override file previews and two of the plugins want to override the same preview? The only other case we have overrides like this is for post components based on the post's type which we provide the best practice of only overriding types specific to your plugin to prevent your plugin negatively affecting other plugins. Is there a best practice we can offer here to avoid two different plugins colliding here (e.g one getting clobbered by the other)? If not, we should try to come up with a way to do this that will allow multiple plugins using this feature to work well together

Reply:
- So, the idea here is that preview component of the first plugin that can override a preview for the given `fileInfo` and `post` objects will be used. 
- This is in a sense similar to the [custom post type component](https://github.com/mattermost/mattermost-webapp/blob/4b3ada33cabcb98719d03f7887aa182f452b3c2a/reducers/plugins/index.js#L160), where if multiple plugins try to register the same post-type, the first plugin component in sorted order of plugin-ids is used.

- Although this may change depending on plugin requirements, one best practice we could suggest plugin authors is to override file previews for specific file types. An example would be that one plugin would override previews for pdf files and another for office files.

Reply:
Not sure there is a lot that can be done here. I hesitate to make this more complicated than it needs to be just to allow multiple plugins to coexist. The best practice suggested that plugins limit themselves to one or a reasonably small set of related extensions seems reasonable.

The only case I worry about here is some kind of "better previews for MM" plugin clashing with "super png previews for MM" plugin. Where I want the second to override the first. The only way I see resolving that is somehow allowing the user to specify which one gets priority. A workaround could be to rename the second `asuperpngpreivews` because then it gets sorted first.

Anyway I think that this might be out of scope for this PR.

Reply:
Agreed that there probably isn't a great solution to this. I'm fine with it as is then but we should clearly document the dangers of overriding all file previews. A comment on the register API saying something like:

```
// Only one plugin can override a file preview at a time. If two plugins try to override the same file preview, the first plugin will perform the override and the second will not. Plugin precedence is ordered alphabetically by plugin ID.
```